apps/web/src/services/dashboards/shared/dashboard-widget-input-form/DashboardWidgetInputForm.vue

```diff
@@ -438,7 +438,10 @@ export default defineComponent<Props>({
         };
 
         /* sync to widget form store */
-        watch(() => state.schemaFormData, (schemaFormData) => {
+        watch(() => state.schemaFormData, (schemaFormData, before) => {
+            if (isEmpty(state.widgetOptionsJsonSchema)) return;
+            if (state.widgetOptionsJsonSchema.properties?.cost_group_by && !schemaFormData?.cost_group_by) return;
+            if (state.widgetOptionsJsonSchema.properties?.asset_group_by && !before?.asset_group_by) return;
```

Comment:
description: I added it to prevent initializing the form when there is no `group by` at the very beginning.